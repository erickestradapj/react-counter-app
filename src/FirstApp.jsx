export const FirstApp = ({ title, subtitle }) => {
   return (
      <>
         <h1>{title}</h1>
         {/* <code>{JSON.stringify(newMessage)}</code> */}
         <p>{subtitle}</p>
      </>
   );
};
