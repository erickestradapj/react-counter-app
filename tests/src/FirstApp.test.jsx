import { render } from '@testing-library/react';
import { FirstApp } from '../../src/FirstApp';

describe('Test in <FirstApp />', () => {
   test('should match with the snapshot', () => {
      const title = 'Hello, I am Goku';
      const { container } = render(<FirstApp title={title} />);

      expect(container).toMatchSnapshot();
   });

   test('should show title in h1', () => {
      const title = 'Hello, I am Goku';
      const { container, getByText } = render(<FirstApp title={title} />);

      expect(getByText(title)).toBeTruthy();

      const h1 = container.querySelector('h1');
      expect(h1.innerHTML).toContain(title);
   });
});
