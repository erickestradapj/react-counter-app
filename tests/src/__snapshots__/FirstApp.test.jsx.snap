// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp /> should match with the snapshot 1`] = `
<div>
  <h1>
     
    Hello, I am Goku
     
  </h1>
  <p>
    No subtitle
  </p>
  <p>
    Erick E.
  </p>
</div>
`;
